### PR TITLE
fix(install): `bun update` keeps pinned versions

### PR DESCRIPTION
**Pull request by dylan-conway**
### What does this PR do?
`bun update` will keep `~` or `^` if it's used in the version. Previously only `^` would stay if used.

Also fixes a bug where `--latest` would not apply to packages passed to `bun update`.

### How did you verify your code works?
added tests

### CODE REVIEW

**Comment by dylan-conway** [comment]
`overrides.test.ts` failure:
Screenshot 2024-06-03 at 3 11 22 PM

File: test/cli/install/bun-update.test.ts [hunk 1]
```diff
@@ -265,7 +265,7 @@ it("should update to latest versions of dependencies", async () => {
     name: "foo",
     dependencies: {
       "@barn/moo": "^0.1.0",
-      baz: "^0.0.5",
+      baz: "~0.0.5",
```
Comment (Jarred-Sumner) [inline test/cli/install/bun-update.test.ts]: Can we run this test for both `~0.0.5` and `^0.0.5`?

File: test/cli/install/bun-update.test.ts [hunk 2]
```diff
@@ -133,7 +133,7 @@ it("should update to latest version of dependency", async () => {
   expect(await file(join(package_dir, "package.json")).json()).toEqual({
     name: "foo",
     dependencies: {
-      baz: "^0.0.5",
+      baz: "~0.0.5",
```
Comment (Jarred-Sumner) [inline test/cli/install/bun-update.test.ts]: Can we run this test for both `~0.0.5` and `^0.0.5`?

Comment (dylan-conway) [reply test/cli/install/bun-update.test.ts]: 👍

File: test/cli/install/bun-update.test.ts (hunk 1 above)
Comment (dylan-conway) [reply test/cli/install/bun-update.test.ts]: 👍

**Pull request merged**